File: Python/lark_card_kit/builders/hr_builder.py
"""Horizontal rule builder for fluent API."""

from __future__ import annotations

from ..models.elements import Hr


class HrBuilder:
    """Horizontal rule builder for fluent API."""

    def __init__(self) -> None:
        """Initialize the HR builder."""
        self._hr = Hr()

    def element_id(self, element_id: str) -> HrBuilder:
        """Set the element ID.

        Args:
            element_id: Element ID.

        Returns:
            Self for method chaining.
        """
        self._hr.element_id = element_id
        return self

    def margin(self, margin: str) -> HrBuilder:
        """Set the margin.

        Args:
            margin: Margin value.

        Returns:
            Self for method chaining.
        """
        self._hr.margin = margin
        return self

    def build(self) -> Hr:
        """Build the horizontal rule element.

        Returns:
            Built Hr instance.
        """
        return self._hr

File: TypeScript/LarkCardKit/src/builders/HrBuilder.ts
import type { Hr } from '../models/elements';

export class HrBuilder {
  private _hr: Hr = { tag: 'hr' };

  elementId(id: string): this {
    this._hr.element_id = id;
    return this;
  }

  margin(margin: string): this {
    this._hr.margin = margin;
    return this;
  }

  build(): Hr {
    return this._hr;
  }
}
